src/devices/src/virtio/queue.rs

```diff
@@ -31,6 +31,8 @@ pub enum QueueError {
     DescIndexOutOfBounds(u16),
     /// Attempted an invalid write into the used ring.
     UsedRing(GuestMemoryError),
+    /// Empty queue
+    EmptyQueue,
```

Comment:
This change defines a specific error at the generic virtio queue level. This clearly breaks modularization and separation of concerns as an empty queue is not a concern for the generic code. 

Ideally we want to be in a position to easily jump to this implmentation: https://github.com/rust-vmm/vm-virtio/blob/master/src/queue.rs/